src/app/components/task-details/task-details.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Router, ParamMap, ActivatedRoute } from '@angular/router';
+import { TaskService } from 'src/app/sevices/task.service';
+import { TaskComponent } from '../task/task.component';
+import { Subscription } from 'rxjs';
+import { Task } from 'src/app/models/task';
+
+@Component({
+  selector: 'app-task-details',
+  standalone: true,
+  imports: [TaskComponent],
+  templateUrl: './task-details.component.html',
+  styleUrl: './task-details.component.scss'
+})
+
+export class TaskDetailsComponent implements OnInit, OnDestroy {
+  paramSubscription: Subscription = Subscription.EMPTY;
+  tasktitle: string | null = null;
+  task: Task | null = null;
+
+  constructor(private router: Router, private activatedRoute: ActivatedRoute, private taskService: TaskService) {}
+
+  ngOnInit(): void {
+    this.paramSubscription = this.activatedRoute.paramMap.subscribe((paramMap: ParamMap) => {
+      this.tasktitle = paramMap.get("title");
+      if(this.tasktitle){
+        this.task = this.taskService.getTaskByTitle(this.tasktitle);
```

Comment:
would be better if you created an ID field, and search by ID since IDs will always be unique, title can repeat